internal/provider/waypoint/resource_waypoint_add_on_definition.go

```diff
@@ -105,26 +106,29 @@ func (r *AddOnDefinitionResource) Schema(ctx context.Context, req resource.Schem
 				},
 			},
 			"readme_markdown_template": schema.StringAttribute{
-				Description: "The markdown template for the Add-on Definition README.",
+				Description: "The markdown template for the Add-on Definition README. Must be base 64 encoded.",
 				Optional:    true,
 			},
 			"terraform_cloud_workspace_details": &schema.SingleNestedAttribute{
-				Required:    true,
-				Description: "Terraform Cloud Workspace details",
+				Optional: true,
+				Computed: true,
+				Description: "Terraform Cloud Workspace details. If not provided, defaults to " +
+					"the HCP terraform project of the associated application.",
 				Attributes: map[string]schema.Attribute{
 					"name": &schema.StringAttribute{
 						Required:    true,
-						Description: "Name of the Terraform Cloud Workspace",
+						Description: "Name of the Terraform Cloud Project",
 					},
 					"terraform_project_id": &schema.StringAttribute{
 						Required:    true,
-						Description: "Tetraform Cloud Project ID",
+						Description: "Terraform Cloud Project ID",
 					},
 				},
 			},
 			"terraform_no_code_module": &schema.SingleNestedAttribute{
-				Required:    true,
-				Description: "Terraform Cloud no-code Module details.",
+				Required: true,
+				Description: "Terraform Cloud no-code Module details. Refer to " +
+					"https://developer.hashicorp.com/terraform/language/modules/sources for more details.",
```

Comment:
I like this, but in reality, the Waypoint server actually only supports a source that is fully quantified right? We don't support local sources for example, so linking them to these docs is a bit misleading.

Reply:
We should still fix this to not link to the docs!